scm-ui/ui-extensions/src/binder.test.ts

```diff
@@ -99,4 +99,21 @@ describe("binder tests", () => {
     expect(extensions[0]).toEqual("planetD");
     expect(extensions[1]).toEqual("planetB");
   });
+
+  it("should allow typings for extension points but still be backwards-compatible", () => {
+    type TestExtensionPointA = ExtensionPointDefinition<"test.extension.a", number, undefined>;
+    type TestExtensionPointB = ExtensionPointDefinition<"test.extension.b", number, { testProp: boolean[] }>;
```

Comment:
You are defining the extension point name inside the type. What about if we have not a fix name like in the markdown renderer? We use like `markdown-renderer.code.${language}`.